packages/hardhat-ethers/src/internal/ethers-utils.ts

```diff
@@ -0,0 +1,459 @@
+// these helpers functions were copied verbatim from ethers
```

Comment:
Functions copied from ethers's codebase, and slightly adapted to comply with our linter. Is this comment enough? Probably not.